Create garatronic pybstick26-esp32c3

garatronic opened the pull request:
<!-- Thanks for submitting a Pull Request! We appreciate you spending the
     time to improve MicroPython. Please provide enough information so that
     others can review your Pull Request.

     Before submitting, please read:
     https://github.com/micropython/micropython/blob/master/CODEOFCONDUCT.md
     https://github.com/micropython/micropython/wiki/ContributorGuidelines

     Please check any CI failures that appear after your Pull Request is opened.
-->

### Summary
Add Garatronic pybstick26-esp32c3 board to micropython deposit


### Testing

Including #17388 issues and correct them.

garatronic committed:
Create board folder for pybstick26-esp32c3 board.

garatronic committed:
Update board.json to correct #17388 issues.

Picture of the board : pull request #39 on micropython-media

dpgeorge commented on ports/esp32/boards/GARATRONIC_PYBSTICK26_ESP32C3/modules/pybstick26_esp32c3.py:
```diff
@@ -0,0 +1,29 @@
+# PYBSTICK26-ESP32C3 MicroPython Helper Library
```
I think it would be better to call this file `board.py` to make it a bit easier to import and use.  Eg:
```py
import board
board.led.on()
```

dpgeorge commented on ports/esp32/boards/GARATRONIC_PYBSTICK26_ESP32C3/modules/pybstick26_esp32c3.py:
```diff
@@ -0,0 +1,29 @@
+# SPI
+SPI_MOSI = const(7)
+SPI_MISO = const(2)
+SPI_CLK = const(6)
```
You shouldn't really need to add any of these pin aliases.  Instead use a `pins.csv` file (see eg `ports/esp32/boards/UM_TINYC6/pins.csv`).  You could do the following:
```csv
SPI_MOSI,GPIO7
SPI_MISO,GPIO2
SPI_CLK,GPIO6

I2C_SDA,GPIO0
I2C_SCL,GPIO1

LED,GPIO8

BUTTON_A,GPIO9
BUTTON_B,GPIO5
```
You can also add other things like `IO6,GPIO6`.  Generally you want to add all aliases for pins that are labelled on the silk screen, and use the silk screen name for the alias.  That way someone looking at the board can know which name to use for each pin.

Then you could do:
```py
led = Pin("LED", Pin.OUT, value=0)
button = Pin("BUTTON_B", Pin.IN, Pin.PULL_UP)
```

garatronic replied:
Yes. Of course. I take the wrong example with LOLIN_C3_MINI and follow it as a mindless.
Thanks for your help.

dpgeorge reviewed (commented):
You don't need to close the PR and create a new one.  You can just add commits to this one that address the review.

garatronic committed:
GARATRONIC_PYBSTICK26_ESP32C3 board support :
- remove useless 'modules' directory, update 'manifest.py'
- add pins.csv for McHobby port notation 'GPx'

garatronic commented:
Hope this will be OK now. Works fine locally. Thanks

dpgeorge commented:
> Hope this will be OK now.

It looks like you didn't commit all of your changes?  You've only added `pins.csv` with the latest commit.

garatronic commented:
I double check local and github deposit. All seems to be OK. I also pull image of the board in micropython-media.

dpgeorge commented:
> I double check local and github deposit. All seems to be OK

You still have the file named `pybstick26_esp32c3.py` file here, and it contains the pin definitions that are no longer needed.

You might need to do `git add `modules/board.py` to add the new file?  Or `git mv modules/pybstick26_esp32c3.py modules/board.py` to rename it via git.

garatronic committed:
Create garatronic pybstick26_esp32c3 support
	modified :        manifest.py
	deleted  :        modules/pybstick26_esp32c3.py

dpgeorge commented on ports/esp32/boards/GARATRONIC_PYBSTICK26_ESP32C3/manifest.py:
```diff
@@ -0,0 +1 @@
+include("$(PORT_DIR)/boards/manifest.py")
```
this file can be removed, it uses the default

garatronic committed:
Remove 'manifest.py' file, useless as set as default
Remove line #9 of mpconfigboard.cmake,useless after removing 'mpconfigboard.cmake' file.

garatronic commented:
Sorry to borrow you. Do you see anything else.

garatronic committed:
Merge branch 'micropython:master' into master

garatronic committed:
'git add mpconfigboard.cmake' file

garatronic committed:
Merge branch 'master' of https://github.com/garatronic/micropython

garatronic commented:
It would be long to explain but ... line 9 of `mpconfigboard.cmake` is removed.

garatronic commented on ports/esp32/boards/GARATRONIC_PYBSTICK26_ESP32C3/board.json:
```diff
@@ -0,0 +1,23 @@
+        "WiFi"
+    ],
+    "images": [
+        "GAR-PYBSTICK26-C3-mUSB-00.JPG"
```
Ok, it seems to be clear for me. Try to do it with ... two push only :)
